airbyte-ci/connectors/pipelines/pipelines/utils.py

```diff
@@ -353,7 +355,7 @@ def get_modified_connectors(modified_files: Set[Path], dependency_scanning: bool
     modified_connectors = set()
     for modified_file in modified_files:
         if not _is_ignored_file(modified_file):
-            modified_connectors.update(_find_modified_connectors(modified_file, dependency_scanning))
+            modified_connectors.update(_find_modified_connectors(modified_file, all_connectors, dependency_scanning))
```

Comment:
@bnchrch the optional dependency caching was already implemented in `_find_modified_connectors`, the calling code was basically always passing `True`